pkg/northbound/gnmi/v2/set.go

```diff
@@ -211,35 +217,52 @@ func (s *Server) Set(ctx context.Context, req *gnmi.SetRequest) (*gnmi.SetRespon
 	return nil, ctx.Err()
 }
 
-func (s *Server) getTargetInfo(ctx context.Context, targets map[configapi.TargetID]*targetInfo, idPrefix string, id configapi.TargetID) (*targetInfo, error) {
+func (s *Server) getTargetInfo(ctx context.Context, targets map[configapi.TargetID]*targetInfo,
+	overrides *configapi.TargetVersionOverrides, idPrefix string, id configapi.TargetID) (*targetInfo, error) {
 	targetID := configapi.TargetID(idPrefix)
 	if len(id) > 0 {
 		targetID = id
 	}
 
+	// Just return the target info if we already have it
 	if target, found := targets[targetID]; found {
 		return target, nil
 	}
 
-	configurable, err := s.getTargetConfigurable(ctx, topoapi.ID(targetID))
-	if err != nil {
-		log.Warn(err)
-		return nil, err
+	// Use the type/version overrides if they are specified
+	var targetType configapi.TargetType
+	var targetVersion configapi.TargetVersion
+	persistent := true // TODO determine whether this is an appropriate default
```

Comment:
I think we can still get configurable for this flag and use it. The exception will be just for target type and version when extension is defined

Reply:
Fixed.